code/modules/mod/mod_core.dm

```diff
@@ -355,3 +355,70 @@
 	add_charge(uses_needed * charge_given)
 	balloon_alert(user, "core refueled")
 	return TRUE
+
+/obj/item/mod/core/plasma/lavaland
+	name = "MOD plasma flower core"
+	icon_state = "mod-core-plasma-flower"
+	desc = "A strange flower from the desolate wastes of lavaland. It pulses with a strange purple glow.  \
+		The wires coming out of it could be hooked into a MODsuit."
+	light_system = MOVABLE_LIGHT
+	light_color = "#cc00cc"
+	light_range = 2
+
+	var/mob/living/mob = /mob/living/basic/butterfly/lavaland/temporary
+	var/max_spawns = 3
+	var/spawned = 0
+	var/can_spawn = TRUE
+	var/children = list()
+
+	// Slightly better than the normal plasma core.
+	// Not super sure if this should just be the same, but will see.
+	maxcharge = 15000
+	charge = 15000
+
+/obj/item/mod/core/plasma/lavaland/Destroy()
+	for(var/mob/child in children)
+		qdel(child)
+	if(mod)
+		if(mod.wearer)
+			mod.wearer.particles = null
+	. = ..()
+
+/obj/item/mod/core/plasma/lavaland/process()
+	if(can_spawn)
+		can_spawn = FALSE
+		addtimer(CALLBACK(src, TYPE_PROC_REF(/obj/item/mod/core/plasma/lavaland, spawn_mob)), 5 SECONDS)
+
+/obj/item/mod/core/plasma/lavaland/proc/child_despawned()
+	spawned--
```

Comment:
Why handle this count separately instead of just using the length of the `children` list?